### DISCUSSION TITLE
StateAdapt documentation

### DISCUSSION THREAD

**Discussion by mailok** [state-adapt/state-adapt#87] [Ideas]
Hello, I wanted to comment on the documentation. I believe this library deserves to have more beautiful documentation. Have you looked into Astro or [Vitepress](https://vitepress.dev/)? I mention these two because I've seen some documentation and they look really good.

Having experience using Vitepress, I felt tempted to do a POC with the documentation, but for that, I needed to have everything in md files, and one thing led to another, and I never did it. In my current company, we decided to leave StoryBook aside to document our components and started using VitePress, and today we are very grateful we took that path. I even used it for my [blog](https://www.tsmailok.dev/), and it was very easy to do. I'll leave you some pages of documentation made with Vitepress so you can get an idea of how they look:

- https://vuejs.org/
- https://vueuse.org/
- https://vitejs.dev/
- https://vitepress.dev/
- https://vitest.dev/

I would be delighted to help build it if you think it's worth it.

**Comment by mfp22** [comment]
I thought about this for a while and thought I responded.

On one hand, I spent a lot of time building these docs, and I have some custom stuff in there to create cheat sheets for GitHub Copilot. But on the other hand, I'm not a designer, and there is something off about the design. But more importantly, it's not SEO-friendly and someone else also wanted search functionality, which I assume would come out of the box with some of these solutions. 

I don't know Vue though. I would feel best about this if we could use Qwik or Astro. 

I'm also not thrilled about docs being really separated from the code itself, but I think I could tolerate it if there was a good system built around it. Everything has to be painstakingly updated for every API change anyway, so this isn't as much an inconvenience as it would be on a regular project.

**Reply by mfp22** [reply]
Maybe Docusaurus would be good. Vitepress, but with React. I think Motion Canvas uses it, and I love those docs. https://docusaurus.io/

Need to figure out how it would deal with my custom stuff though.

**Reply by pBouillon** [reply]
Just a suggestion but it seems that Angular is finally getting somewhere with content driven websites with [Analog](https://analogjs.org/) (as hinted by Joshua Morony on [his latest video](https://www.youtube.com/watch?v=fMtwsJNGQCU)). It uses Vite, SSR and can feed on markdown content. It could be an alternative worth to consider if the custom stuff you are mentionning is Angular related.

 **EDIT:** There is even an [integration with Asto](https://analogjs.org/docs/packages/astro-angular/overview)

**Comment by mfp22** [comment]
I looked more thoroughly into this. I clicked around those Vitepress websites and for a little bit I was tempted to stay on Angular because the initial page load is about the same, but subsequent navigations take more than twice as long on the Vitepress websites. Even though I have carefully optimized what is able to be lazy loaded, something like Vitepress should be performing better out of the box, and it's kind of disappointing to see how slow navigations are. It's not terrible; it's just a little disappointing.

Qwik City with client routing was my personal preference if performance was all that mattered, but I don't think 2.0 is out yet, and even if it is coming out soon, I'm sick of maintaining the technology behind the documentation. I'm craving something standardized.

And for me to upgrade to Angular 16 (yeah this repo is stuck that far back, although I test it on more modern repos) I have to upgrade from carbon-elements-angular v4 to v5, which totally redoes the CSS, which is the main thing I was originally fighting. There's something appealing to me about that design, but it's not worth how much customization I've had to add on top of it. And the fact that it's built on web components and not for Angular makes it less than optimal. Even with that, the performance was better, in my opinion, than those other sites. So it's mostly the maintenance that comes from overoptimizing the CSS that bothers me (by default it loads ALL of it up-front). I basically have to re-implement all the basic stuff anyway, so it may be a good time to try something new.

And Vitepress looks so convenient. It's built with documentation in mind. I'll have to adjust hopefully almost none of the CSS, and little things like anchor links just look great out of the box.

Anyway, I'm going to upgrade the repo to Nx 19 this week, and at that point probably start working on a Vitepress version of the documentation. But I'll make the final call when I get to that point.

**Reply by mfp22** [reply]
Okay for a bit I was leaning towards Docusaurus, but the thing that pushed me back to Vitepress was Vite over Webpack, and also the Nx plugin I found to add Vitepress was more recent than the one I found for Docusaurus. 

But I got it working in Nx and so hopefully I can quickly get the docs moved over to this new thing. But I have a lot of work to do. 
![image](https://github.com/user-attachments/assets/75597aff-c230-4250-8537-79e28f5f8a40)

It's going to be nice to have some docs that look good by default.

**Comment by mfp22** [comment]
Done https://state-adapt.github.io/
Will open issues for bugs
